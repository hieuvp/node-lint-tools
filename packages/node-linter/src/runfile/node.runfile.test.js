const lint = require('./node.runfile');
const { eslint } = require('./eslint.runfile');
const { jsonlint } = require('./jsonlint.runfile');
const { prettier } = require('./prettier.runfile');

jest.mock('./eslint.runfile', () => ({ eslint: jest.fn() }));
jest.mock('./jsonlint.runfile', () => ({ jsonlint: jest.fn() }));
jest.mock('./prettier.runfile', () => ({ prettier: jest.fn() }));

const validatedArgs = [
  ['.'],
  [
    'packages/hapi-linter/',
    'packages/node-linter/',
    'packages/react-linter/',
    'packages/react-native-linter/'
  ],
  ['runfile.js', '.eslintrc.js'],
  ['runfile.js', '.eslintrc.js', 'packages/'],
  ['runfile.js', 'packages/']
];

/**
 * Set default Jest working directory to $ProjectFileDir$ (absolute path)
 * if you are using WebStorm and want to run a single test
 */

describe('args validation', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  validatedArgs.forEach(args => {
    it(`should be happy because ${JSON.stringify(args)} is a validated args`, () => {
      expect(() => lint(...args)).not.toThrow();
    });
  });

  it('should throw an error if args is empty', () => {
    expect(() => lint()).toThrowErrorMatchingSnapshot();
  });

  it('should only accept one args if contains a wildcard (.) otherwise will throw error', () => {
    expect(() => lint('.', 'src')).toThrowErrorMatchingSnapshot();
  });

  ['src', 'test/', 'config.js'].forEach(args => {
    it(`should throw an error because "${args}" does not exist`, () => {
      expect(() => lint(args)).toThrowErrorMatchingSnapshot();
    });
  });

  ['package.json', 'node_modules', 'packages/node-linter/node_modules/'].forEach(args => {
    it(`should throw an error because "${args}" is in the blacklist`, () => {
      expect(() => lint(args)).toThrowErrorMatchingSnapshot();
    });
  });
});
