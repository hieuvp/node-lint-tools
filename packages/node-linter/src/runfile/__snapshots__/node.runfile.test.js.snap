// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`args validation should only accept one args if contains a wildcard (.) otherwise will throw error 1`] = `"Wildcard (.) is already covered everything, there's no need to define more arguments"`;

exports[`args validation should throw an error because "config.js" does not exist 1`] = `"ENOENT: no such file or directory, stat 'config.js'"`;

exports[`args validation should throw an error because "node_modules" is in the blacklist 1`] = `"Path \\"node_modules/\\" is blacklisted"`;

exports[`args validation should throw an error because "package.json" is in the blacklist 1`] = `"Path \\"package.json\\" is blacklisted"`;

exports[`args validation should throw an error because "packages/node-linter/node_modules/" is in the blacklist 1`] = `"Path \\"packages/node-linter/node_modules/\\" is blacklisted"`;

exports[`args validation should throw an error because "src" does not exist 1`] = `"ENOENT: no such file or directory, stat 'src'"`;

exports[`args validation should throw an error because "test/" does not exist 1`] = `"ENOENT: no such file or directory, stat 'test/'"`;

exports[`args validation should throw an error if args is empty 1`] = `"Please specify patterns or files"`;
